sdk/core/azure-core/test/perf/inc/azure/core/test/http_transport_test.hpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ * @brief Test the HTTP send performance.
+ *
+ */
+
+#pragma once
+
+#include "../../../core/perf/inc/azure/perf.hpp"
+
+#include <azure/core.hpp>
+#include <azure/core/http/curl_transport.hpp>
+#include <azure/core/http/win_http_transport.hpp>
+
+#include <memory>
+
+namespace Azure { namespace Core { namespace Test {
+
+  /**
+   * @brief Measure the HTTP transport performance.
+   */
+  class HTTPTransportTest : public Azure::Perf::PerfTest {
+
+    std::string m_target;
+    std::shared_ptr<Azure::Core::Http::HttpTransport> m_Transport;
+    Azure::Core::Http::HttpMethod m_httpMethod = Azure::Core::Http::HttpMethod::Get;
+
+    void GetRequest()
+    {
+      auto httpRequest = Azure::Core::Http::Request(m_httpMethod, Azure::Core::Url(m_target));
+      Azure::Core::Context context;
+      auto response = m_Transport->Send(httpRequest, context);
+      // Make sure to pull all bytes from network.
+      auto body = response->ExtractBodyStream()->ReadToEnd();
+    }
+
+    void PostRequest()
+    {
+      std::string payload = "{}";
+      Azure::Core::IO::MemoryBodyStream payloadStream(
+          reinterpret_cast<const uint8_t*>(payload.data()), payload.size());
+      auto httpRequest
+          = Azure::Core::Http::Request(m_httpMethod, Azure::Core::Url(m_target), &payloadStream);
+      Azure::Core::Context context;
+      auto response = m_Transport->Send(httpRequest, context);
+      // Make sure to pull all bytes from network.
+      auto body = response->ExtractBodyStream()->ReadToEnd();
+    }
+
+  public:
+    /**
+     * @brief Construct a new HTTPTransportTest test.
+     *
+     * @param options The test options.
+     */
+    HTTPTransportTest(Azure::Perf::TestOptions options) : PerfTest(options) {}
+
+    void Setup() override
+    {
+      if ("winhttp" == m_options.GetMandatoryOption<std::string>("Transport"))
+      {
+#if defined(BUILD_TRANSPORT_WINHTTP_ADAPTER)
```

Comment:
The "winhttp" option shouldn't be available if winhttp isn't being built. Same is true for libcurl.

Alternatively, if the customer configures winhttp and it's not available, fail the test. And if curl isn't available and they don't specify winhttp, fail the test.

Reply:
There is no customer, i am the customer, made changes as requested, just to clarify we are the customer, noone else.